packages/util-dynamodb/src/DocumentClient.ts

```diff
@@ -0,0 +1,101 @@
+ * | Boolean                                                                | BOOL                    |
+ * | null                                                                   | NULL                    |
+ * | Array                                                                  | L                       |
+ * | Object                                                                 | M                       |
+ * | Buffer, File, Blob, ArrayBuffer, DataView, and JavaScript typed arrays | B                       |
+ *
+ */
+export class DocumentClient extends DynamoDB {
+  /**
+   * Creates a new item, or replaces an old item with a new item by
+   * delegating to `DynamoDB.putItem()`.
+   *
+   * Supply the same parameters as {DynamoDB.putItem} with
+   * `AttributeValue`s substituted by native JavaScript types.
+   *
+   * @see DynamoDB.putItem
+   * @example Create a new item in a table
+   *  var params = {
+   *    TableName : 'Table',
+   *    Item: {
+   *       HashKey: 'haskey',
+   *       NumAttribute: 1,
+   *       BoolAttribute: true,
+   *       ListAttribute: [1, 'two', false],
+   *       MapAttribute: { foo: 'bar'},
+   *       NullAttribute: null
+   *    }
+   *  };
+   *
+   *  var documentClient = new DocumentClient();
+   *
+   *  try {
+   *   const data = await documentClient.put(params);
+   *   console.log(data);
+   *  } catch(err) {
+   *   console.log(err);
+   *  }
+   *
+   */
+  public put(args: DocumentPutInput, options?: __HttpHandlerOptions): Promise<DocumentPutOutput>;
+  public put(args: DocumentPutInput, cb: (err: any, data?: DocumentPutOutput) => void): void;
+  public put(
+    args: DocumentPutInput,
+    options: __HttpHandlerOptions,
+    cb: (err: any, data?: DocumentPutOutput) => void
+  ): void;
+  public put(
+    args: DocumentPutInput,
+    optionsOrCb?: __HttpHandlerOptions | ((err: any, data?: DocumentPutOutput) => void),
+    cb?: (err: any, data?: DocumentPutOutput) => void
+  ): Promise<DocumentPutOutput> | void {
+    // Do input translation on args, and send translated args to PutItemCommand
+    // @ts-ignore: TypeScript throws "Property 'TableName' is missing"
+    const command = new PutItemCommand({ ...args, Item: marshall(args.Item) });
+
+    // Create new callback to perform output translation on second value of cb or optionsOrCb
+    const getCallBack = (callback: (err: any, data?: DocumentPutOutput) => void) => (
+      err: any,
+      data?: PutItemCommandOutput
+    ) => {
+      callback(err, { ...data, Attributes: unmarshall(data.Attributes) });
+    };
+
+    if (typeof optionsOrCb === "function") {
+      this.send(command, getCallBack(optionsOrCb));
+    } else if (typeof cb === "function") {
+      if (typeof optionsOrCb !== "object") throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
+      this.send(command, optionsOrCb || {}, getCallBack(cb));
+    } else {
+      // @ts-ignore
+      return this.send(command, optionsOrCb);
```

Comment:
Where does the unmarshalling happen here?

Reply:
Looks like [here](https://github.com/aws/aws-sdk-js-v3/pull/2031/files#diff-6660574b0011d37413395db8878ef4673df4bed3a5d70b9580971a083b1ed9abR88)

Reply:
My bad, I hadn't pushed local commits.
The unmarshalling was added in `Promise.resolve()` in [`6f6e793` (#2031)](https://github.com/aws/aws-sdk-js-v3/pull/2031/commits/6f6e7933eb9ca18395aca334c231cacbc10fba6d)

Reply:
It was pushed seconds after the question;-)